### PR TITLE
Add transaction refund api method

### PR DESCRIPTION
**Pull request by prakashpp**

### CODE REVIEW

File: transaction.py [hunk 1]
```diff
@@ -708,6 +721,26 @@ def get_shipping_address(self, name):
         """
         return None
 
+    def create_refund(self, amount=None):
+        refund_transaction, = self.copy([self])
+
+        refund_transaction.type = 'refund'
+        refund_transaction.amount = amount or self.amount
+        refund_transaction.origin = self
+        refund_transaction.provider_reference = self.provider_reference
```
Comment (sharoonthomas) [inline transaction.py]: refund_transaction.provider_reference = None

File: transaction.py [hunk 2]
```diff
@@ -209,6 +215,11 @@ class PaymentTransaction(Workflow, ModelSQL, ModelView):
         states=READONLY_IF_NOT_DRAFT, depends=['state'],
         required=True, select=True
     )
+    last_four_digits = fields.Char('Last Four Digits')
```
Comment (sharoonthomas) [inline transaction.py]: this needs a migration (manually for our users is also fine) where we can take the last_four_digits from profile for most cases.

File: transaction.py [hunk 3]
```diff
@@ -116,6 +116,12 @@ class PaymentTransaction(Workflow, ModelSQL, ModelView):
         'Description', states=READONLY_IF_NOT_DRAFT,
         depends=['state']
     )
+    type = fields.Selection(
```
Comment (sharoonthomas) [inline transaction.py]: Add to the view ?

File: transaction.py (hunk 1 above)
Comment (prakashpp) [inline transaction.py]: done

File: transaction.py (hunk 2 above)
Comment (prakashpp) [inline transaction.py]: wrote the migration.

File: transaction.py (hunk 3 above)
Comment (prakashpp) [inline transaction.py]: added

**Pull request merged**